### DISCUSSION TITLE
Keystone keeps redirecting to /signin

### DISCUSSION THREAD

**Discussion by elsvenno98** [keystonejs/keystone#7924] [Questions]
I have the following problem:

- When I use keystone on localhost I can make calls from my Angular client with Apollo to the api without any problems.

- When I deploy my keystone application to my dev environment and did not login yet, my api calls all get redirected to /signin from the admin ui. But when i login ones in the browser my calls will go through to the api instead of bein redirected to /signin.
- When I deploy my app with Ionic i face the problem of it always getting redirected to /signin. I've tried a lot to first make a call to the api for a token, but even then it keeps redirecting me to /signin. I even tried sending requests to /signin but even then it redirects me to /signin. Is there anyway to work around this problem or is there something that i did do wrong while configuring my keystone?

**Comment by flexdinesh** [comment]
Have you setup auth or are you using keystone without auth?

Here's an example on how to setup auth with Keystone — https://github.com/keystonejs/keystone/blob/main/examples/auth/keystone.ts

If you've setup auth, you should use the `authenticateUserWithPassword` in the generated GraphQL API to authenticate and use the token in your subsequent API requests.

**Reply by elsvenno98** [reply]
Yes I've set keystone up with Auth.

When i want to get a token from the API I send the authenticateUserWithPassword mutation to the api. But then the API directly redirects my request to /signin so it doesn't do anything with the request.

I've put the mutation that I'm sending to 'baseurl/api/graphql' down below.
[authenticate_user_mutation.txt](https://github.com/keystonejs/keystone/files/9563522/authenticate_user_mutation.txt)

**Reply by flexdinesh** [reply]
That shouldn't happen. Could you share an example code you've setup in a codesandbox or github repo?

**Reply by elsvenno98** [reply]
I've copied my gitlab repo to github. It is at the following link, removed all company specific info like deployment files.
https://github.com/elsvenno98/keystonedebug/tree/development
